extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AddNamespaceDecorator.java

```diff
@@ -18,10 +20,19 @@ public AddNamespaceDecorator(String namespace) {
     }
 
     @Override
-    public void andThenVisit(ObjectMetaBuilder builder, ObjectMeta resourceMeta) {
-        if (!builder.hasNamespace()) {
-            builder.withNamespace(namespace);
-        }
+    public void visit(KubernetesListBuilder list) {
+        List<HasMetadata> buildItems = list.buildItems()
+                .stream()
+                .peek(obj -> {
+                    if (obj instanceof Namespaced) {
+                        final ObjectMeta metadata = obj.getMetadata();
+                        if (metadata.getNamespace() == null) {
```

Comment:
I forgot to add a comment here: this is meant to keep the previous functionality of not overriding a manually set namespace, i.e. the namespace provided via `quarkus.kubernetes.namespace` should only be applied to resources that don't already specify one.